eng/mgmt/gulpfile.js

```diff
@@ -107,9 +112,10 @@ var codegen = function(project, cb) {
 
     console.log('Generating "' + project + '" from spec file ' + specRoot + '/' + mappings[project].source);
 
-    const generatorPath = args['autorest-java']
-        ? `--use=${path.resolve(args['autorest-java'])} `
-        : '';
+    const generatorPath = autoRestJavaVersion == 'preview' || autoRestJavaVersion == 'latest' 
+		|| autoRestJavaVersion.match(/[0-9]+\.[0-9]+\.[0-9]+.*/)
```

Comment:
This could match some path include version, like ~/test-1.2.3/***. May use ^ at first, but I think split into two parameters is better

Reply:
@xccc-msft please check tanyi's suggestion. I think use ^ would be fine.

If possible also print out the resolution of this line, if not already there. Then if anything went wrong caller would have log to see why.